devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,412 @@
+			}
+			tileAttribute("device.displayName", key: "SECONDARY_CONTROL") {
+				attributeState "displayName", label: 'Model:  ${currentValue}'
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel:false, decoration:"flat", width:6, height:2) {
+			state "battery", label:'${currentValue}% BATTERY', unit:""
+		}
+		main "toggle"
+		details(["toggle", "battery"])
+	}
+}
+
+// Globals - Cluster IDs
+private getCLUSTER_POWER() { 0x0001 }
+private getCLUSTER_DOORLOCK() { 0x0101 }
+private getCLUSTER_ALARM() { 0x0009 }
+
+// Globals - Command IDs
+private getDOORLOCK_CMD_UNLOCK_DOOR() { 0x1F }
+private getDOORLOCK_RESPONSE_OPERATION_EVENT() { 0x20 }
+private getPOWER_ATTR_BATTERY_VOLTAGE() { 0x0020 }
+private getDOORLOCK_ATTR_LOCKSTATE() { 0x0000 }
+private getDOORLOCK_ATTR_DOORSTATE() { 0x0003 }
+private getALARM_ATTR_ALARM_COUNT() { 0x0000 }
+private getALARM_CMD_ALARM() { 0x00 }
+
+/**
+ * Called on app installed
+ */
+def installed() {
+	log.trace "ZigBee DTH - Executing installed() for device ${device.displayName}"
+}
+
+/**
+ * Called on app uninstalled
+ */
+def uninstalled() {
+	def deviceName = device.displayName
+	log.trace "ZigBee DTH - Executing uninstalled() for device $deviceName"
+	sendEvent(name: "lockRemoved", value: device.id, isStateChange: true, displayed: false)
+}
+
+
+/**
+ * Ping is used by Device-Watch in attempt to reach the device
+ */
+def ping() {
+	log.trace "ZigBee DTH - Executing ping() for device ${device.displayName}"
+	refresh()
+}
+
+/**
+ * Called when the user taps on the refresh button
+ */
+def refresh() {
+	log.trace "ZigBee DTH - Executing refresh() for device ${device.displayName}"
+	def cmds = zigbee.readAttribute(CLUSTER_DOORLOCK, DOORLOCK_ATTR_DOORSTATE)
+	log.info "ZigBee DTH - refresh() returning with cmds:- $cmds"
+	return cmds
+}
+
+
+/**
+ * Configures the device to settings needed by SmarthThings at device discovery time
+ *
+ */
+def configure() {
+	log.trace "ZigBee DTH - Executing configure() for device ${device.displayName}"
+	/**
+	* Configure Reporting is not set for Samsung Smart Doorlock devices as the doorlocks are programmed to automatically report their status
+	* Lock state is automatically reported every 30 minutes
+	* Battery state is automatically reported every 12 hours
+	* Battery state is also reported when the batteries are exchanged
+	*/
+	def cmds = zigbee.command(0x0000, 0x1E, "",[mfgCode: 0003]) // read modelName of the device as it is not being sent with zbjoin.
+	sendEvent(name: "lock", value: "unlocked", isStateChange: true, displayed: false)
+	sendEvent(name: "battery", value: "100", isStateChange: true, displayed: false)
+	// Device-Watch allows 2 check-in misses from device + ping (plus 2 min lag time)
+	// Currently set for testing ping interval set to 3 min
+	sendEvent(name: "checkInterval", value: 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
You don't want to keep this at 3 minutes. If it's just for testing then you should change your local DTH. But the version that goes into the repo should have the real value (1h2m would be my suggestion).

This will destroy the battery.

Reply:
Yes, the final version should have this updated. I was going to put this back at the setting of the original ZigBee Lock DTH(2h 2m). Please advise me on setting this value for deployment.

Reply:
2h 2m is good.